test/Repository/index.test.js

```diff
@@ -101,4 +102,43 @@ describe('Repository', () => {
             'https://github.com/all-contributors/all-contributors-bot/pull/1347',
         )
     })
+
+    test('createPullRequest with files but a branch or pull request already exists', async () => {
+        nock('https://api.github.com')
+            .get(
+                `/repos/all-contributors/all-contributors-bot/git/refs/heads/master`,
+            )
+            .reply(200, gitGetRefdata)
+
+        nock('https://api.github.com')
+            .post(
+                `/repos/all-contributors/all-contributors-bot/git/refs`,
+                verifyBody,
+            )
+            .reply(422)
+
+        const error = await rejectionOf(
+            repository.createPullRequestFromFiles({
```

Comment:
When we are creating the pull request and the branch exists can we recover somehow?

E.g. delete the branch then create. Or Maybe have a unique I'd for the branch we are creating (using UUID)?

Reply:
I prefer the second suggestion.

Reply:
Create a unique branch so that it is the easiest and the smallest to change

Reply:
If the branch is already open, I think that the bot should be able to (re)-use it.
If the PR is already open then it shouldn't be doing anything **unless** it's changing/adding/removing something else.